x/evm/abci.go

```diff
@@ -10,23 +10,23 @@ import (
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
 )
 
-// BeginBlock sets the Bloom and Hash mappings and resets the Bloom filter and
+// BeginBlock sets the block hash -> block height map and resets the Bloom filter and
 // the transaction count to 0.
 func BeginBlock(k Keeper, ctx sdk.Context, req abci.RequestBeginBlock) {
 	if req.Header.LastBlockId.GetHash() == nil || req.Header.GetHeight() < 1 {
 		return
 	}
 
-	// Consider removing this when using evm as module without web3 API
-	bloom := ethtypes.BytesToBloom(k.Bloom.Bytes())
-	k.SetBlockBloomMapping(ctx, bloom, req.Header.GetHeight())
-	k.SetBlockHashMapping(ctx, req.Header.LastBlockId.GetHash(), req.Header.GetHeight())
+	k.SetBlockHash(ctx, req.Header.LastBlockId.GetHash(), req.Header.GetHeight()-1)
```

Comment:
is it possible to set the block hash in `EndBlock` as well, instead of setting the previous block hash here?

Reply:
you can't because you only have a reference to the last block, not the current one being processed. We want to set the block hash during begin block so it's available during tx/query execution. Remember the ABCI block lifecycle from the SDK talk: `EndBlock` -> (for each Tx) `DeliverTx` -> `EndBlock` -> `Commit`.

Reply:
ah i see, that makes sense. thanks for explaining :D